packages/babel-core/src/transformation/file/options/build-config-chain.js

```diff
@@ -107,17 +115,29 @@ class ConfigChainBuilder {
 
     this.resolvedConfigs.push(loc);
 
-    let content = fs.readFileSync(loc, "utf8");
     let options;
-
-    try {
-      options = jsonCache[content] = jsonCache[content] || json.parse(content);
-      if (key) options = options[key];
-    } catch (err) {
-      err.message = `${loc}: Error while parsing JSON - ${err.message}`;
-      throw err;
+    if (path.extname(loc) === ".js") {
+      try {
+        let configModule = require(loc);
+        // Check for `exports.default` in case babel-register is being used
+        // and the .babelrc.js file contains an `export default`.
+        options = configModule.default || configModule;
```

Comment:
We've been pretty consistent about only using `.default` if `__esModule`, probably best to keep it consistent.

```
options = configModule && configModule.__esModule ? configModule.default : configModule;
```